codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/CborSerializerGenerator.kt

```diff
@@ -267,7 +280,16 @@ class CborSerializerGenerator(
     }
 
     override fun unsetStructure(structure: StructureShape): RuntimeType =
-        UNREACHABLE("Only clients use this method when serializing an `@httpPayload`. No protocol using CBOR supports this trait, so we don't need to implement this")
+        ProtocolFunctions.crossOperationFn("cbor_json_unset_struct_payload") { fnName ->
```

Comment:
Is this change to `unsetStructure` essential for initial response functionality, or just a nice-to-have improvement? If it's necessary for initial responses, shouldn't we similarly update the `unsetUnion` function which has the same "UNREACHABLE" implementation?